Content.Client/GameObjects/Components/Mobs/ClientStatusEffectsComponent.cs

```diff
@@ -80,32 +85,74 @@ private void PlayerDetached()
             _ui = null;
         }
 
-        public void UpdateIcons()
+        public void UpdateStatusEffects()
         {
             if (!CurrentlyControlled || _ui == null)
             {
                 return;
             }
+            _cooldown.Clear();
             _ui.VBox.DisposeAllChildren();
 
-            foreach (var effect in _icons.OrderBy(x => (int) x.Key))
+            foreach (var effect in _status.OrderBy(x => (int) x.Key))
             {
-                TextureRect newIcon = new TextureRect
+                var cooldown = new CooldownGraphic();
+
+                Control status = new Control()
                 {
-                    TextureScale = (2, 2),
-                    Texture = _resourceCache.GetTexture(effect.Value)
+                    Children =
+                    {
+                        new TextureRect
+                        {
+                            TextureScale = (2, 2),
+                            Texture = _resourceCache.GetTexture(effect.Value.Icon)
+                        },
+                        cooldown,
+                    }
                 };
 
-                newIcon.Texture = _resourceCache.GetTexture(effect.Value);
-                _ui.VBox.AddChild(newIcon);
+                _cooldown[effect.Key] = cooldown;
+
+                _ui.VBox.AddChild(status);
             }
         }
 
         public void RemoveIcon(StatusEffect name)
```

Comment:
Should this keep this name?

Reply:
In the end I changed it to be more accurate. `RemoveStatusEffect`